src/octopeer-github/main/Settings/Settings.ts

```diff
@@ -0,0 +1,106 @@
+/**
+ * Created by Mitchell on 12-5-2016.
+ */
+
+/**
+ * Class from which user settings can be retrieved.
+ * These settings have been maintained by the front-end.
+ */
+class Settings {
```

Comment:
Nooonononono :P I bet typescript has much better ways of creating singletons, I haven't seen much of it, but I'd first research that before using `getInstance()` :)